android-project/app/src/main/res/xml/backup_descriptor.xml

```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<full-backup-content>
+    <exclude
+        domain="sharedpref"
+        path="hidapi.xml" />
+</full-backup-content>

```

Comment:
This is interesting. Do you know how this backup feature works?

I read that it backs up several folders, including the external storage directory, and that backups are limited to 25 MB per user of the app. However, I don't know what happens when a backup exceeds 25 MB, as ours inevitably would even with just spawn.mpq. It would be great if save files and INI settings could be restored upon reinstall so it would be a bit disappointing if the whole backup process fails because of the MPQ files.

Reply:
Not really, after updating to 31 Android Studio highlighted the `android:allowBackup="true"` line in `AndroidManifest.xml` and said that it could generate the now needed `backup_descriptor.xml`.

Looking at the docs the path only supports exact match, so we can't do `*.sv` :(

I guess the solution is to have it be `.` and exclude all known MPQ files.

Reply:
Hrm, you're right, and of course it's case-sensitive, too. It feels absurd to me that they don't support wildcards, but I guess they can justify keeping it simple because they allow overriding the Auto Backup process with a BackupAgent written in Java.

Excluding the known MPQ files should be good enough for us, though. 👍

Reply:
I split this up in to two PRs so that we can get backup working before we do an upgrade to SDK 31